src/main/java/com/example/bookstore/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,145 @@
+import com.example.bookstore.model.Order;
+import com.example.bookstore.model.Order.Status;
+import com.example.bookstore.model.OrderItem;
+import com.example.bookstore.model.ShoppingCart;
+import com.example.bookstore.model.User;
+import com.example.bookstore.repository.OrderItemRepository;
+import com.example.bookstore.repository.OrderRepository;
+import com.example.bookstore.repository.ShoppingCartRepository;
+import com.example.bookstore.repository.UserRepository;
+import com.example.bookstore.service.OrderService;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.EnumSet;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final OrderMapper orderMapper;
+    private final OrderItemMapper orderItemMapper;
+    private final UserRepository userRepository;
+    private final ShoppingCartRepository shoppingCartRepository;
+
+    @Override
+    public List<OrderDto> findAllOrdersByUser(String email, Pageable pageable) {
+        return orderRepository.findAllOrderByUserId(getUserByEmail(email).getId(), pageable)
+                .stream()
+                .map(orderMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public OrderItemDto getOrderItemById(String email, Long orderId, Long itemId) {
+        return getOrderByUserId(email, orderId)
+                .getOrderItems()
+                .stream()
+                .filter(orderItem -> orderItem.getId().equals(itemId))
+                .findFirst()
+                .map(orderItemMapper::toDto).orElseThrow(
+                        () -> new EntityNotFoundException("Can't find item with id: " + itemId)
+                );
+    }
+
+    @Override
+    public OrderDto createNewOrder(String email, CreateOrderRequestDto dto) {
+        User user = getUserByEmail(email);
+        ShoppingCart shoppingCart = shoppingCartRepository.findByUserWithItems(user.getEmail())
+                .orElseThrow(() ->
+                        new EntityNotFoundException("Can't find shopping cart by user id: "
+                                + user.getId()));
+        Order order = new Order();
+        order.setUser(user);
+        order.setStatus(Status.PENDING);
+        order.setTotal(countTotal(shoppingCart));
+        order.setOrderDate(LocalDateTime.now());
+        order.setShippingAddress(dto.shippingAddress());
+        Set<OrderItem> orderItems = createOrderItems(order,shoppingCart.getCartItems());
+        order.setOrderItems(orderItems);
+        orderRepository.save(order);
+        orderItemRepository.saveAll(orderItems);
```

Comment:
it would have been more handy to use cascades here instead of saving them in 2 repo calls, but that is up to you :)